From 4e18fa39e51331a7f34e402ef9fb14a71325e2f0 Mon Sep 17 00:00:00 2001
From: Liwei Song <liwei.song@windriver.com>
Date: Fri, 3 Jan 2020 10:19:20 +0800
Subject: [PATCH] allow mount to dirs under /tmp

fix avc denials:

AVC avc:  denied  { mounton } for  pid=646 comm="mount" path="/tmp/mnt"
dev="tmpfs" ino=26671 scontext=root:sysadm_r:mount_t:s0-s15:c0.c1023
tcontext=root:object_r:user_tmp_t:s0 tclass=dir permissive=0

Upstream-Status: Pending

Signed-off-by: Liwei Song <liwei.song@windriver.com>
Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/system/mount.te      |  4 ++++
 policy/modules/system/userdomain.if | 18 ++++++++++++++++++
 2 files changed, 22 insertions(+)

diff --git a/policy/modules/system/mount.te b/policy/modules/system/mount.te
index 422d411e3..1fbd467b4 100644
--- a/policy/modules/system/mount.te
+++ b/policy/modules/system/mount.te
@@ -47,6 +47,10 @@ allow mount_t self:capability { chown dac_override ipc_lock setgid setuid sys_ad
 allow mount_t mount_tmp_t:file manage_file_perms;
 allow mount_t mount_tmp_t:dir manage_dir_perms;
 
+
+# Allow mount to dirs under /tmp
+userdom_mounton_tmp_dirs(mount_t)
+
 can_exec(mount_t, mount_exec_t)
 
 files_tmp_filetrans(mount_t, mount_tmp_t, { file dir })
diff --git a/policy/modules/system/userdomain.if b/policy/modules/system/userdomain.if
index f2771594a..9d433c37c 100644
--- a/policy/modules/system/userdomain.if
+++ b/policy/modules/system/userdomain.if
@@ -3430,6 +3430,24 @@ interface(`userdom_mounton_user_runtime_dirs',`
 	allow $1 user_runtime_t:dir mounton;
 ')
 
+#######################################
+## <summary>
+##  Mount a filesystem on user tmp dir
+## </summary>
+## <param name="domain">
+##  <summary>
+##  Domain allowed access.
+##  </summary>
+## </param>
+#
+interface(`userdom_mounton_tmp_dirs',`
+	gen_require(`
+		type user_tmp_t;
+	')
+
+	allow $1 user_tmp_t:dir mounton;
+')
+
 ########################################
 ## <summary>
 ##	Relabel to user runtime directories.
-- 
2.17.1

